### PR TITLE
Fully qualify register/field macro names.

### PR DESCRIPTION
**Pull request by timsifive**
This solves the problem we have with the same field existing in multiple
registers, as reported in #451.

### CODE REVIEW

File: debug_module.tex [hunk 1]
```diff
@@ -54,55 +54,55 @@ \section{Debug Module Interface (DMI)} \label{dmi}
...
 These bits must be set regardless of the cause of the reset.
 The {\tt havereset} bits for the selected harts
-can be cleared by writing 1 to \Fackhavereset in \Rdmcontrol.
+can be cleared by writing 1 to \FdmDmcontrolAckhavereset in \RdmDmcontrol.
 The {\tt havereset} bits may or may not be cleared
-when \Fdmactive is low.
+when \FdmDmcontrolDmactive is low.
 
-When a hart comes out of reset and \Fhaltreq or \Fresethaltreq are set, the
+When a hart comes out of reset and \FdmDmcontrolHaltreq or \Fresethaltreq are set, the
```
Comment (ernie-sifive) [inline debug_module.tex]: How did you decide which registers and fields were changed to the fully qualified name and which weren't?  `\Fhaltreq` was changed but `\Fresethaltreq` was not.  And `\Fhartsel` vs. `\FdmDmcontrolHasel`

**Comment by jscheid-ventana** [comment]
Is the intent to namespace the field names, or just fix the document linking?  If the former, I suggest using the <registername>.<fieldname> convention found in the priv spec.

File: debug_module.tex (hunk 1 above)
Comment (timsifive) [reply debug_module.tex]: I changed every macro that was autogenerated.

resethaltreq isn't a bit in a register. These bits exists somewhere in the DM, and are accessed by dmcontrol.setresethaltreq and dmcontrol.clrresethaltreq.

hartsel is similar. It's the combination of dmcontrol.hartsello and dmcontrol.hartselhi.

**Comment by timsifive** [comment]
> Is the intent to namespace the field names, or just fix the document linking? If the former, I suggest using the . convention found in the priv spec.

For now just to fix the linking. If we want to refer to each field as register.field we can make that change easily in registers.py.

**Pull request merged**